minor optimizations for nft wallet tests

trepca opened the pull request.

trepca committed:
minor optimizations for nft wallet tests

trepca committed:
minor optimizations for nft wallet tests

ChiaMineJP committed:
Improve CI performance (#12523)

* Optimized CI execution times

* Fixed a lint error

* Update GUI cache keys for linux

* Added `git status` and `git submodule` to build_xxx-installer.sh

xdustinface committed:
util: Unify lockfile handling  (#12541)

* util: Unify lockfile handling

This patch:
- Renames `util.lock.lock_by_path` to `util.lock.lock_path` and improves it by using `FileLock`
- Unifies all places where we create lock files to use `chia.util.lock.lock_path`
- Changes the daemon lockfile from `start-daemon.launching` to `daemon.lock` like we have it with all `<service>.lock` in the `run` folder
- Drops the `fastener` dependency

* Move the not keyring related sync tests into `test_lockfile.py`

xdustinface committed:
harvester: Log the plot refresh parameter on start (#12661)

arvidn commented on tests/wallet/nft_wallet/test_nft_wallet.py:
```diff
@@ -1293,15 +1198,10 @@ async def test_set_nft_status(two_wallet_nodes: Any, trusted: Any) -> None:
+    await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
+    await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
 
-    await time_out_assert(20, wallet_0.get_unconfirmed_balance, funds)
```
we don't care about asserting the unconfirmed balance anymore?

trepca replied:
we're just trying to figure out wallet is ready for minting and there are no transactions happening as we're minting first few blocks before (previous code was mainly copy/paste from tests for standard wallet afair)

arvidn committed:
log blockchain database file used along with schema version on startup (#12737)

ytx1991 committed:
Add bulk cancel API (#12427)

* Add bulk cancel API

* Add cancel_all flag

* Add asset ID

* Resolve comments

* Fix pre-commit

Commit:
rename testnet constants to default constants, because they are the base for all networks (#12691)

* mild annoyance

* Update default_constants.py

Commit:
add extra messages to daemon (#12666)

add extra messages to daemon

add if for sanity

Oops pycharm

Commit:
Fix no keys loaded error by making KeychainProxy automatically reconnect when a connection is lost (#12782)

* Add proper types to daemon client & keychain

* pre type pass on keychain proxy

* add auto restarting keychain proxy

* finalize keychain proxy v2

* move event calls

* fix close on timeout

* fix possible key leak in logging

* fix last little error

* log on disconnection & reduce duplicate code

* Update client.py

Quexington committed:
Block summaries of CAT1 offers in the RPC and CLI (#12730)

* Block summaries of CAT1 offers in the RPC

* block CLI too

* manually decompress/deserialize

* black

* Add test

* pre-commit

emlowe committed:
New test files with CAT2 coins (#12683)

* New test files with CAT2 coins

* Remove unused test file

arvidn committed:
uncurry the puzzle once instead of 3 times (#12781)

Commit:
Simulator Final touches (#12703)

* add new rpc and add tests.

I added, get_all_blocks rpc, and expanded the get_all_puzzle_hashes rpc.

* lint pass

* update plot directory for simulator

* return peak height on new block farm

* expand test coverage

* fix plot directory craziness

* simulator sync override

* Add new simulator networktype

* add logging to simulator

* Fix race condition, mempool and confirm

* add reorg seed

* farm block if block 0 sim & fix unsynced for sim

* lint

* oops i forgot to lint

* switch to detecting simulator in network type

also remove inline genesis generation

* fix config loading to use correct variable

Co-authored-by: Mariano Sorgente <sorgente711@gmail.com>

xdustinface committed:
util: Move all keychain related errors into `chia.util.error` (#12756)

Also derive them from `KeychainException` and name them consistently.

Commit:
fix missing import

screw git

trepca committed:
increased timeouts

ChiaMineJP committed:
Fixed an issue where it sometimes tried to remove a non-existent win cert file (#12797)

emlowe committed:
Remove unused ping/pong code (#12779)

Commit:
Move simulator tests over to new file, to allow other programs to use it (#12759)

* Separate test file at Cameron's request

this lets other tools easily use these tests

* fix accidental text duplication

mariano54 committed:
No network type (#12783)

* No network type

* Use selected network

* remove NETWORK_TYPE

* Don't warn on invalid key, and remove from config

* Remove bytes32

* Put back the warning log

xdustinface committed:
rpc: Improve RPC server start/stop (#12768)

* rpc: Improve RPC server start/stop

* `RpcServer.stop` -> `RpcServer.close`

xdustinface committed:
util: Drop partially implemented BIP39 passphrase support (#12748)

arvidn committed:
fix a few instances of coin name logging that would print binary-escaped strings instead of hex (#12818)

trepca committed:
merged

trepca committed:
reorg fix
